integ-test/src/test/java/org/opensearch/sql/sql/TextTypeIT.java

```diff
@@ -0,0 +1,270 @@
+    }
+
+    @Test
+    public void selectPositionText() throws IOException {
+      var result = executeQuery(String.format("select POSITION(\\\"text\\\" IN typeText) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("typeText", null, "double"));
+
+//      }
+    }
+
+    @Test
+    public void selectPositionTextKeywordFieldNoFieldData() {
+      var result = executeJdbcRequest(String.format("select POSITION(\\\"keyword\\\" IN typeKeywordFieldNoFieldData) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("LOCATE('keyword', typeKeywordFieldNoFieldData)", null, "double"));
+    }
+
+    @Test
+    public void selectPositionTypeTextFieldData() throws IOException {
+      var result = executeQuery(String.format("select POSITION(\\\"keyFD\\\" IN typeTextFieldData) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("LOCATE('keyFD', typeTextFieldData)", null, "double"));
+    }
+
+    @Test
+    public void selectPositionTextDataFieldNoFields() {
+      var result = executeJdbcRequest(String.format("select POSITION(\\\"textFDNF\\\" IN textDataFieldNoFields) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("LOCATE('textFDNF', textDataFieldNoFields)", null, "double"));
+    }
+
+    // Substring
+
+    @Test
+    public void selectSubstringKeyword() {
+      var result = executeJdbcRequest(String.format("select SUBSTRING(typeKeyword, 1, 1) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("SUBSTRING(typeKeyword, 1, 1)", null, "keyword"));
+    }
+
+    @Test
+    public void selectSubstringText() {
+      var result = executeJdbcRequest(String.format("select SUBSTRING(typeText, 1, 1) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("SUBSTRING(typeText, 1, 1)", null, "keyword"));
+    }
+
+    @Test
+    public void selectSubstringTextKeywordFieldNoFieldData() {
+      var result = executeJdbcRequest(String.format("select SUBSTRING(typeKeywordFieldNoFieldData, 1, 1) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("SUBSTRING(typeKeywordFieldNoFieldData, 1, 1)", null, "keyword"));
+    }
+
+    @Test
+    public void selectSubstringTypeTextFieldData() {
+      var result = executeJdbcRequest(String.format("select SUBSTRING(typeTextFieldData, 1, 1) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("SUBSTRING(typeTextFieldData, 1, 1)", null, "keyword"));
+    }
+
+    @Test
+    public void selectSubstringTextDataFieldNoFields() {
+      var result = executeJdbcRequest(String.format("select  SUBSTRING(textDataFieldNoFields, 1, 1) from %s", TEST_INDEX_TEXTKEYWORD));
+      verifySchema(result,
+          schema("SUBSTRING(textDataFieldNoFields, 1, 1)", null, "keyword"));
+    }
+
+//    protected JSONObject executeQuery(String query) throws IOException {
```

Comment:
remove commented code?

Reply:
Resolved in 1371d40f65c5e93364abed6e88855c721fe253bf.